### PR TITLE
RedHat based distros now point to SaltStack corporate repo. 

### PR DESCRIPTION
**Pull request by grep4linux**
Added SLE 12 support. Fixed OpenSUSE.

### CODE REVIEW

File: bootstrap-salt.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh -
+h!/bin/sh -
```
Comment (s0undt3ch) [inline bootstrap-salt.sh:1]: h?

**Comment by s0undt3ch** [comment]
I need to review the rest locally since I already have the new repo stuff in a branch on my fork.

File: bootstrap-salt.sh [hunk 2]
```diff
@@ -4457,6 +4469,9 @@ install_opensuse_stable_deps() {
     # shellcheck disable=SC2086
     __zypper_install ${__PACKAGES} || return 1
 
+    # Fix for OpenSUSE 13.2 and 2015.8 - gcc should not be required. Work around until package is fixed by SuSE
+    _EXTRA_PACKAGES="${_EXTRA_PACKAGES} gcc python-devel libgit2-devel"
```
Comment (s0undt3ch) [inline bootstrap-salt.sh:4473]: Why do we need these? Specially pygit2 which is not a hard dependency for Salt.

What type of installation triggered this? Stable packages? Git based installation?

File: bootstrap-salt.sh [hunk 3]
```diff
@@ -4624,8 +4639,200 @@ install_opensuse_check_services() {
...
+    return 0
+}
+
+install_suse_12_stable() {
+    if [ "$SUSE_PATCHLEVEL" -gt 1 ]; then
+        install_opensuse_stable || return 1
+    else
+        # USE_SETUPTOOLS=1 To work around
+        # error: option --single-version-externally-managed not recognized
+        USE_SETUPTOOLS=1 pip install salt || return 1
```
Comment (s0undt3ch) [inline bootstrap-salt.sh:4770]: The openSUSE stable packages don't work in SuSE? Why are we pip installing here?

File: bootstrap-salt.sh (hunk 1 above)
Comment (grep4linux) [inline bootstrap-salt.sh:1]: oops! Sorry about that. keep the normal shebang.

File: bootstrap-salt.sh (hunk 2 above)
Comment (grep4linux) [inline bootstrap-salt.sh:4473]: We looked at this with Tom. He said that it looked like the SUSE salt-minion rpm was requiring git event though we don't need it.  I'll have to re-create the problem, but one of the dependencies, (I think git 2) is being required by SUSE. However, it has a dependency that requires a compiler. My plan is to look into it more, submit a bug to SUSE. I've met the packager Olaf when I worked there and wanted to send him a note.  

All I know at this point is that the only way it works is with those packages added. This is just a standard rpm-based installation.

**Comment by jfindlay** [comment]
@grep4linux, @s0undt3ch, any updates here?

**Comment by clarkperkins** [comment]
Would be nice if this could get merged, as `bootstrap-salt.sh` does not currently work with CentOS 6 due to the bad version of python-crypto.

**Comment by jfindlay** [comment]
@clarkperkins, try https://bootstrap.saltstack.com/develop.

**Comment by vutny** [comment]
@grep4linux Hey, looks like your branch is outdated and requires some rebase/rework.
Is this still an issue after all? As I see it, bootstrapping on Suse has nothing to do with SaltStack corporate repo.

**Comment by jfindlay** [comment]
@vutny, I have talked with @grep4linux and am working on making the changes you suggest and some others.

**Comment by vutny** [comment]
Thank you @jfindlay. Just wanted to say that looks like this PR is no longer relevant and could be closed.

**Pull request closed without merging**